src/test/java/unit/GcsBucketReferenced.java

```diff
@@ -345,5 +391,143 @@ void updateMultipleOrNoProperties() throws IOException {
         TestCommand.runAndParseCommandExpectSuccess(
             UFGcsBucket.class, "resource", "describe", "--name=" + newName);
     assertEquals(newDescription, describeBucket.description);
+    assertEquals(externalPrivateBucket.getName(), describeBucket.bucketName);
+  }
+
+  @Test
+  @DisplayName(
+      "Attempt to update the gcs bucket while the user only have access to the externalSharedBucket")
+  void updateGcsBucketWithPartialAccess() throws IOException {
+    workspaceCreator.login();
+
+    UFWorkspace createWorkspace =
+        TestCommand.runAndParseCommandExpectSuccess(UFWorkspace.class, "workspace", "create");
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+    // `terra workspace add-user --email=$email --role=READER`
+    TestCommand.runCommandExpectSuccess(
+        "workspace", "add-user", "--email=" + shareeUser.email, "--role=WRITER");
+
+    String name = "updateGcsBucketWithPartialAccess";
+    TestCommand.runCommandExpectSuccess(
+        "resource",
+        "add-ref",
+        "gcs-bucket",
+        "--name=" + name,
+        "--bucket-name=" + externalSharedBucket.getName());
+
+    shareeUser.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+
+    String newName = "updateGcsBucketWithPartialAccess_NEW";
+    UFGcsBucket updateBucket =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFGcsBucket.class,
+            "resource",
+            "update",
+            "gcs-bucket",
+            "--name=" + name,
+            "--new-name=" + newName);
+    assertEquals(newName, updateBucket.name);
+
+    TestCommand.runCommandExpectExitCode(
+        1,
+        "resource",
+        "update",
+        "gcs-bucket",
+        "--name=" + name,
+        "--new-bucket-name=" + externalPrivateBucket.getName());
+
+    // clean up
+    TestCommand.runCommandExpectSuccess("resource", "delete", "--name=" + newName, "--quiet");
+    workspaceCreator.login();
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + createWorkspace.id);
+    TestCommand.runCommandExpectSuccess("workspace", "delete", "--quiet");
+  }
+
+  @Test
+  @DisplayName(
+      "Attempt to descrone the gcs bucket while the user does not have access to the externalPrivateBucket")
+  void describeBucketReferenceWithoutAccess() throws IOException {
+    workspaceCreator.login();
+
+    UFWorkspace createWorkspace =
```

Comment:
ditto above, don't create workspaces in test methods because they won't be cleaned up on failure. if the different workspace roles are important here (I'm not totally clear on whether the READER/WRITER distinction is what's causing failures or the private/shared bucket access is), then I suggest making a new test class that creates 2 workspaces in setup (e.g. `workspaceSharedWithReader`, `workspaceSharedWithWriter`) and shares them with a user.